spyder/utils/programs.py

```diff
@@ -30,11 +30,27 @@ class ProgramError(Exception):
     pass
 
 
-if os.name == 'nt':
-    TEMPDIR = tempfile.gettempdir() + osp.sep + 'spyder'
-else:
-    username = encoding.to_unicode_from_fs(getuser())
-    TEMPDIR = tempfile.gettempdir() + osp.sep + 'spyder-' + username
+def get_temp_dir(suffix=None):
+    """
+    Returns the temporary path, checking previously that this path exists.

```

Comment:
Please change this to

    Return temporary Spyder directory, checking previously that it exists.